core/src/main/java/org/jruby/ir/targets/Bootstrap.java

```diff
@@ -657,7 +653,10 @@ public static IRubyObject ivarGet(VariableSite site, IRubyObject self) throws Th
         VariableAccessor accessor = realClass.getVariableAccessorForRead(site.name());
 
         // produce nil if the variable has not been initialize
-        MethodHandle nullToNil = self.getRuntime().getNullToNilHandle();
```

Comment:
The purpose of getNullToNilHandle was to have a single place to cache the handle and reuse it everywhere. This is going to recreate handles every time. I agree we need a better place than on Ruby, but we still need these one-time handles.